angular-ngc/defs.bzl

```diff
@@ -268,30 +274,74 @@ def ng_library(name, package_name = None, deps = [], test_deps = [], visibility
 
     if len(test_spec_srcs) > 0:
         _unit_tests(
-            name = "test",
+            name = "tests",
             tests = test_spec_srcs,
             deps = [":_lib"] + test_deps + TEST_DEPS,
             visibility = visibility,
         )
 
 def _unit_tests(name, tests, deps, visibility):
+    generate_test_setup(name = "test_setup")
+    test_srcs = tests + ["test_setup.ts"]
+
     ng_project(
-        name = "_tests",
-        srcs = tests,
+        name = name,
+        srcs = test_srcs,
         deps = deps,
+        visibility = visibility,
+    )
+
+    karma_polyfills_name = "test_polyfills"
+    generate_karma_polyfills(
+        name = karma_polyfills_name,
+    )
+
+    test_entry = [":%s" % karma_polyfills_name]
```

Comment:
The bootstraping is one file so we should have a single entry that produces a single bundle .js instead of a directory

Reply:
done.